awesomplete.js

```diff
@@ -211,14 +216,19 @@ _.prototype = {
 			this.ul.innerHTML = "";
 
 			this.suggestions = this._list
-				.filter(function(item) {
-					return me.filter(item, value);
+				.map(function(item) {
+					var suggestion = new Suggestion(convert(item));
+					var data = me.data(suggestion, value);
```

Comment:
We can make data method optional and skip this and the following line if it is null/undefined.